backend/baseclass.pm

```diff
@@ -2,6 +2,46 @@
 # Copyright 2012-2021 SUSE LLC
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+# Helper class to identify file descriptors in error case, see do_capture()
+package NamedIOSelect;
+use Mojo::Base -base;
+use IO::Select;
+use Carp;
+
+sub select {
+    return shift->{select_obj} //= IO::Select->new();
```

Comment:
For new code we should better use signatures right away.

Reply:
done